plugins/com.google.cloud.tools.eclipse.usagetracker.test/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManagerTest.java

```diff
@@ -178,10 +153,62 @@ public void testSendPingScheduled_optInRegisteredAndAlreadyOptedIn() {
 
   private void verifyPingQueued(VerificationMode verificationMode) {
     when(pingEventQueue.isEmpty()).thenReturn(true);
-    AnalyticsPingManager pingManager =
-        new AnalyticsPingManager(preferences, display, pingEventQueue, true);
-    pingManager.unitTestMode = true;
     pingManager.sendPing("eventName", "metadataKey", "metadataValue");
     verify(pingEventQueue, verificationMode).add(any(PingEvent.class));
   }
+
+  @Test
+  public void testGetAnonymizedClientId_generateNewId() {
+    when(preferences.get(eq(AnalyticsPreferences.ANALYTICS_CLIENT_ID), anyString()))
+        .thenReturn(null);  // Simulate that client ID has never been generated.
+    String clientId = pingManager.getAnonymizedClientId();
+    assertTrue(!clientId.isEmpty());
+    verify(preferences, times(1)).put(eq(AnalyticsPreferences.ANALYTICS_CLIENT_ID), eq(clientId));
+  }
+
+  @Test
+  public void testGetAnonymizedClientId_useSavedId() {
+    when(preferences.get(eq(AnalyticsPreferences.ANALYTICS_CLIENT_ID), anyString()))
+        .thenReturn("some-unique-client-id");
+    String clientId = pingManager.getAnonymizedClientId();
+    assertEquals("some-unique-client-id", clientId);
+    verify(preferences, never()).put(eq(AnalyticsPreferences.ANALYTICS_CLIENT_ID), eq(clientId));
+  }
+
+  @Test
+  public void testSendPing_validArguments() {
+    pingManager.sendPing("eventName", "metadataKey", "metadataValue");
+    // Should not throw any exception.
```

Comment:
I'll change the method name. 

> You also could verify if pingEventQueue.add() has been called

Enqeueing depends on two opt-in states, and this is actually tested thoroughly in other tests. I'll change the test name accordingly to reflect that this is only verifying arguments.